test/sampling.jl

```diff
@@ -387,6 +387,51 @@ end
   @test length(ps) > 0
 end
 
+@testitem "FibonacciSampling" setup = [Setup] begin
+  @test_throws ArgumentError sample(Box(cart(0, 0), cart(1, 1)), FibonacciSampling(-1))
+  @test_throws ArgumentError sample(Box(Point(0, 0, 0), Point(1, 1, 1)), FibonacciSampling(100))
+
+  box = Box(cart(1, 1), cart(4, 2))
+  ps = sample(box, FibonacciSampling(100)) |> collect
+  @test first(ps) isa Point
+  @test ps[1].coords ≈ cart(1, 1).coords
+  @test all(∈(box), ps)
+
+  box = Box(cart(0, 0), cart(1, 1))
+  ps = sample(box, FibonacciSampling(100, π)) |> collect
+  @test first(ps) isa Point
+  @test all(∈(box), ps)
+  @test ps[2] ≈ cart(mod(1 / π, 1), 1 / 99)
+
+  tbox = Box(cart(0, 0), cart(1, 1))
+  af = Affine([1 1; 0 1], [2, 0])
+  tbox = af(tbox)
+  ps = sample(tbox, FibonacciSampling(100)) |> collect
+  @test first(ps) isa Point
+  @test ps[1].coords ≈ af(cart(0, 0)).coords
+  @test all(∈(tbox), ps)
+
+  disk = Disk(Plane(cart(3, 0, 0), Vec(1, 0, 0)), T(2))
+  ps = sample(disk, FibonacciSampling(100)) |> collect
+  @test first(ps) isa Point
+  @test ps[1].coords ≈ centroid(disk).coords
+  @test all(p -> p.coords.x ≈ 3u"m", ps)
+  @test all(p -> -2u"m" < p.coords.y || p.coords.y < 2u"m" || isapprox(p.coords.y, 2u"m"; atol=1e-5u"m"), ps)
+  @test all(p -> -2u"m" < p.coords.z || p.coords.z < 2u"m" || isapprox(p.coords.z, 2u"m"; atol=1e-5u"m"), ps)
```

Comment:
@davibarreira please refactor p.coords as coords(p)

The only place in the entire stack where we access fields of structs is in the CRS, in which case we type c.x, c.y, c.lat, c.lon, ...

Reply:
I did it so that we can wrap this up quickly.